kernel/Kconfig

```diff
@@ -156,7 +156,7 @@ config SCHED_CPU_MASK_PIN_ONLY
 config MAIN_STACK_SIZE
 	int "Size of stack for initialization and main thread"
 	default 2048 if COVERAGE_GCOV
-	default 1024 if TEST_ARM_CORTEX_M
+	default 2048 if TEST_ARM_CORTEX_M
```

Comment:
This increases the minimum RAM requirement for all Cortex-M targets and can potentially make the tests that used to be able to build for the lower end targets no longer fit in them.

> Increase the stack size used for ARM during testing to cover the
no optimizations test cases.

Why not just set `CONFIG_MAIN_STACK_SIZE=2048` in the tests that build with `CONFIG_NO_OPTIMIZATIONS=y`?

Reply:
Ok. Updated to increase it in the cases affected.